Realizar un modelo lineal para el siguiente problema de predicción de demanda de energía 

| Velocidad del viento | Personas dentro del edificio | Requerimiento de energía |
|-----------------------|------------------------------|---------------------------|
| 100                   | 2                            | 5                         |
| 50                    | 42                           | 25                        |
| 45                    | 31                           | 22                        |
| 60                    | 35                           | 18                        |


In [ ]:
#  Hacer un modelo lineal con la tabla , pero usando la matriz θ = (X^T X) ^ -1  X^T y
import numpy as np

def linear_regression(X, y):
    # Añadir un sesgo (bias) a la matriz de características
    X_b = np.c_[np.ones((X.shape[0], 1)), X]  
    # Calcular θ usando la fórmula θ = (X^T X) ^ -1  X^T y
    theta = np.linalg.inv(X_b.T.dot(X_b)).dot(X_b.T).dot(y)
    return theta

In [2]:
# Probar en la tabla 
X = np.array([[2, 100], [42, 50], [31, 45], [35,60]])
y = np.array([5, 25, 22, 18])
theta = linear_regression(X, y)
print("Parámetros del modelo:", theta)

Parámetros del modelo: [24.97615881  0.20920152 -0.20751687]


Por lo que el modelo queda de la siguiente manera :

$ \hat{y}= \theta_{0} + \theta_{1} x_{1} + \theta_{2} x_{2}$

$ \hat{y}=$ 24.98−0.207⋅(viento)+0.209⋅(personas)